ocs_ci/ocs/resources/storage_cluster.py

```diff
@@ -2669,3 +2650,56 @@ def validate_serviceexport():
     assert mon_count == len(
         get_mon_pods()
     ), f"Mon serviceexport count mismatch {mon_count} != {len(get_mon_pods())}"
+
+
+def get_storage_size():
+    """
+    Get the storagecluster storage size
+
+    Returns:
+        str: The storagecluster storage size
+
+    """
+    sc = get_storage_cluster()
+    storage = (
+        sc.get()
+        .get("items")[0]
+        .get("spec")
+        .get("storageDeviceSets")[0]
+        .get("dataPVCTemplate")
+        .get("spec")
+        .get("resources")
+        .get("requests")
+        .get("storage")
+    )
+    if storage.isdigit or config.DEPLOYMENT.get("local_storage"):
+        # In the case of UI deployment of LSO cluster, the value in StorageCluster CR
+        # is set to 1, so we can not take OSD size from there. For LSO we will return
+        # the size from PVC.
+        pvc = get_deviceset_pvcs()[0]
+        return pvc.get()["status"]["capacity"]["storage"]
+    else:
+        return storage
+
+
+def resize_osd(new_osd_size):
+    """
+    Resize the OSD(e.g., from 512 to 1024, 1024 to 2048, etc.)
+
+    Args:
+        new_osd_size (str): The new osd size(e.g, 512Gi, 1024Gi, 1Ti, 2Ti, etc.)
+
+    Returns:
+        bool: True in case if changes are applied. False otherwise
+
+    """
+    sc = get_storage_cluster()
```

Comment:
please add a check that the new requested size is greater than the existing OSD size.
And should we add a test case of tyring to resize to a smaller size?

Reply:
Okay, np. 
We added a negative test in the test plan: https://docs.google.com/spreadsheets/d/1wCSwMk_L4SikzZ2UarUH1-VCXyGkHs-ZNdDEtWPG7f4/edit#gid=105186681&range=B24

Reply:
do you plan to add it in a follow up PR?

Reply:
I have implemented the osd size check. 
I will add the negative test in another PR. I need to check first manually what is the behavior